Mage.Tests/src/test/java/org/mage/test/player/TestPlayer.java

```diff
@@ -2652,6 +2652,38 @@ public int getAmount(int min, int max, String message, Game game) {
         return computerPlayer.getAmount(min, max, message, game);
     }
 
+    @Override
+    public List<Integer> getMultiAmount(int amount, List<String> messages, Game game) {
+        assertAliasSupportInChoices(false);
+        int size = messages.size();
+        int choiceAmount = 0;
+        boolean allValuesChosen = true;
+        List<Integer> multiAmount = new ArrayList<>(size);
+        for (String message : messages) {
+            if (choices.isEmpty() || !choices.get(0).startsWith("X=")) {
+                this.chooseStrictModeFailed("choice", game, message);
```

Comment:
XMage uses AI choices on missing commands from the dev. So `chooseStrictModeFailed` must be called before AI usage only. You call assert error on wrong command.

Choose dialog structure:
* process with dev commands and choices;
  * if you found wrong commands then show error;
  * if all fine then return;
* process with AI:
  * if it strict mode then show error

There are a special choice `TestPlayer.CHOICE_SKIP`. Look at `addTargetAmount` and `TestPlayer.TARGET_SKIP` for example. You must use same logic for multi choices. Example:
* mana dialog with 5 mana to choose;
* you want to use second mana in the unit test;
* two commands list possible:
  1.
```
  setChoice(playerA, "X=0");
  setChoice(playerA, "X=1");
  setChoice(playerA, "X=0");
  setChoice(playerA, "X=0");
  setChoice(playerA, "X=0");
```
  2. 
```
  setChoice(playerA, "X=0");
  setChoice(playerA, "X=1");
  setChoice(playerA, TestPlayer.CHOICE_SKIP);
```

If dev forget to add all choice commands then test must fail. User auto-fill choices only for skip command.

Reply:
If you need help with AI and unit tests then tell me -- I'll add unit tests and improve test framework for your new dialog here (after it finished).